test/gnmi/topotest.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gnmi
+
+import (
+	"context"
+	"github.com/onosproject/onos-api/go/onos/topo"
+	"github.com/onosproject/onos-config/test/utils/gnmi"
+	toposdk "github.com/onosproject/onos-ric-sdk-go/pkg/topo"
+	testutils "github.com/onosproject/onos-test/pkg/utils"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+// NewTopoClient creates a new topo SDK client
+func NewTopoClient() (toposdk.Client, error) {
+	return toposdk.NewClient()
+}
+
+// getEntityOrDie queries an entity of the given kind and fails if it isn't found
+func getEntityOrDie(ctx context.Context, t *testing.T, c toposdk.Client, kind string) topo.Object {
+	filter := &topo.Filters{
+		KindFilter: &topo.Filter{
+			Filter: &topo.Filter_Equal_{Equal_: &topo.EqualFilter{Value: kind}},
+		}}
+	entities, err := c.List(ctx, toposdk.WithListFilters(filter))
+	assert.NoError(t, err)
+	assert.Len(t, entities, 1)
+	return entities[0]
+}
+
+// getOnosConfigEntityOrDie queries a single onos-config instance and fails if it isn't found
+func getOnosConfigEntityOrDie(ctx context.Context, t *testing.T, c toposdk.Client) topo.Object {
+	return getEntityOrDie(ctx, t, c, topo.ONOS_CONFIG)
+}
+
+// getDevicesimEntityOrDie queries a single device simulator instance and fails if it isn't found
+func getDevicesimEntityOrDie(ctx context.Context, t *testing.T, c toposdk.Client) topo.Object {
+	const (
+		devicesimKind = "Devicesim"
+	)
+	return getEntityOrDie(ctx, t, c, devicesimKind)
+}
+
+// getRelationOrDie queries a single control relation between the config and target, and fails if not found
+func getRelationOrDie(ctx context.Context, t *testing.T, configID topo.ID, targetID topo.ID, c toposdk.Client) topo.Object {
+	filter := &topo.Filters{
+		RelationFilter: &topo.RelationFilter{
+			SrcId:    string(configID),
+			TargetId: string(targetID),
+			Scope:    topo.RelationFilterScope_ALL,
+		},
+		KindFilter: &topo.Filter{
```

Comment:
I don't think you need this. Simply set RelationKind field int the RelationFilter to topo.CONTROLS.